service-api/module/Application/src/Yoti/YotiService.php

```diff
@@ -123,12 +116,196 @@ public function retrieveResults(string $sessionId): array
+                'headers' => $headers,
+                'query' => [
+                    'sdkId' => $this->sdkId->getValue(),
+                    'sessionId' => $yotiSessionId,
+                    'nonce' => $nonce,
+                    'timestamp' => $timestamp
+                ]
+            ]);
+
+            if ($config->getStatusCode() !== Response::STATUS_CODE_200) {
+                $this->logger->error('Case configuration retrieval was unsuccessful ', [
+                    'data' => [ "sessionId" => $yotiSessionId]
+                ]);
+                throw new YotiException("Error: " . $config->getReasonPhrase());
+            }
+            return json_decode(strval($config->getBody()), true);
+        } catch (GuzzleException $e) {
+            $this->logger->error('Unable to connect to Yoti service [' . $e->getMessage() . '] ', [
+                'data' => [ "operation" => "session-configuration", "sessionId" => $yotiSessionId ]
+            ]);
+            throw new YotiException("A connection error occurred. Previous: " . $e->getMessage());
+        }
+    }
+
+    /**
+     * @param CaseData $caseData
+     * @return array
+     * PDF Stage 2: Retrieve PDF letter contents
+     * @throws YotiException
+     */
+    public function retrieveLetterPDF(CaseData $caseData, string $nonce, int $timestamp): array
+    {
+        $headers = $this->getSignedRequest(
+            '/sessions/' . $caseData->sessionId . '/instructions/pdf',
+            'GET',
+            $nonce,
+            $timestamp,
+            null,
+            $caseData->sessionId
+        );
+        try {
+            $pdfData = $this->client->get('/idverify/v1/sessions/' . $caseData->sessionId . '/instructions/pdf', [
+                'headers' => $headers,
+                'query' => [
+                    'sdkId' => $this->sdkId->getValue(),
+                    'sessionId' => $caseData->sessionId,
+                    'nonce' => $nonce,
+                    'timestamp' => $timestamp
+                ],
+            ]);
+        } catch (GuzzleException $e) {
+            $this->logger->error('Unable to connect to Yoti service [' . $e->getMessage() . '] ', [
+                'data' => [ 'operation' => 'retrieve-pdf', 'header' => $headers]
+            ]);
+            throw new YotiException("A connection error occurred. Previous: " . $e->getMessage());
+        }
+        $base64 = base64_encode(strval($pdfData->getBody()));
+        // Convert base64 to pdf
+        $pdf = base64_decode($base64);
+
+        return ["status" => "PDF Created", "pdfData" => $pdf];
+    }
+
+    public function letterConfigPayload(CaseData $caseData, string $requirementId): array
+    {
+        $payload = [];
+
+        $payload["contact_profile"] = [
+            "first_name" => $caseData->firstName,
+            "last_name" => $caseData->lastName,
+            "email" => 'opg-all-team+yoti@digital.justice.gov.uk'
```

Comment:
Worth making this email an environment var so it's not hard coed and we can change across environments.